pages/perfil/index.public.js

```diff
@@ -140,21 +141,23 @@ function EditProfileForm() {
       if (response.status === 200) {
         await fetchUser();
 
+        clearTimeout(sucessTimeoutRef.current);
+
+        setGlobalSucessMessage('Salvo com sucesso!');
+
+        sucessTimeoutRef.current = setTimeout(() => {
+          setGlobalSucessMessage(null);
+        }, 5000);
+
         if (user.email !== email) {
+          setGlobalSucessMessage(null);
```

Comment:
Não acha melhor colocar tudo isso após o `else` ?

```suggestion
        if (user.email !== email) {
```


```js
    ...
    } else {
        clearTimeout(sucessTimeoutRef.current);

        setGlobalSucessMessage('Salvo com sucesso!');

        sucessTimeoutRef.current = setTimeout(() => {
          setGlobalSucessMessage(null);
        }, 5000);
    }